src/gnssro/gnssro_AWSopendataNetcdf2ioda.py

```diff
@@ -0,0 +1,495 @@
+    VarAttrs[('sequenceNumber', 'MetaData')]['units'] = '1'
+
+    VarAttrs[('satelliteIdentifier', 'MetaData')]['longname'] = 'Low Earth Orbit satellite identifier, e.g., COSMIC2=750-755'
+    VarAttrs[('satelliteIdentifier', 'MetaData')]['units'] = '1'
+
+    VarAttrs[('satelliteTransmitterId', 'MetaData')]['longname'] = 'GNSS satellite transmitter identifier'
+    VarAttrs[('satelliteTransmitterId', 'MetaData')]['units'] = '1'
+
+    VarAttrs[('satelliteAscendingFlag', 'MetaData')]['flag_values'] = '0,1'
+    VarAttrs[('satelliteAscendingFlag', 'MetaData')]['flag_meanings'] = 'descending/ascending'
+    VarAttrs[('satelliteAscendingFlag', 'MetaData')]['longname'] = 'the original occultation ascending/descending flag'
+    VarAttrs[('satelliteAscendingFlag', 'MetaData')]['units'] = '1'
+
+    VarAttrs[('dataProviderOrigin', 'MetaData')]['longname'] = 'originally data processing_center,e.g., 60=UCAR'
+    VarAttrs[('dataProviderOrigin', 'MetaData')]['name_in_earlier_version'] = 'process_center'
+    VarAttrs[('dataProviderOrigin', 'MetaData')]['units'] = '1'
+
+    VarAttrs[('satelliteConstellationRO', 'MetaData')]['longname'] = 'GNSS satellite classification, e.g., 401=GPS'
+    VarAttrs[('satelliteConstellationRO', 'MetaData')]['units'] = '1'
+
+    VarAttrs[('impactHeightRO', 'MetaData')]['units'] = 'meter'
+    VarAttrs[('impactHeightRO', 'MetaData')]['_FillValue'] = float_missing_value
+    VarAttrs[('impactHeightRO', 'MetaData')]['valid_range'] = '0-200000'
+    VarAttrs[('impactHeightRO', 'MetaData')]['longname'] = 'distance from mean sea level'
+
+    VarAttrs[('impactParameterRO', 'MetaData')]['units'] = 'meter'
+    VarAttrs[('impactParameterRO', 'MetaData')]['_FillValue'] = float_missing_value
+    VarAttrs[('impactParameterRO', 'MetaData')]['valid_range'] = '6200000-6600000'
+    VarAttrs[('impactParameterRO', 'MetaData')]['longname'] = 'centre of curvature'
+
+    VarAttrs[('bendingAngle', 'ObsValue')]['units'] = 'Radians'
+    VarAttrs[('bendingAngle', 'ObsValue')]['_FillValue'] = float_missing_value
+    VarAttrs[('bendingAngle', 'ObsValue')]['valid_range'] = '0-0.008'
+    VarAttrs[('bendingAngle', 'ObsValue')]['longname'] = 'Bending Angle'
+
+    # final write to IODA file
+    writer.BuildIoda(obs_data, VarDims, VarAttrs, GlobalAttrs)
+    sys.exit()
+
+
+def read_input(input_file_and_record, add_qc):
+    """
+    Reads/converts input file(s)
+
+    Arguments:
+
+        input_args: an input filename or names
+            input_file: The name of file to read
+
+    Returns:
+
+        A dictionary holding the variables (obs_data) needed by the IODA writer
+    """
+
+    input_file = input_file_and_record[0]
+    record_number = input_file_and_record[1]
+    print("Reading: %s" % input_file)
+
+    f = h5py.File(input_file, 'r')
+    obs_data = get_opendata(f, add_qc, record_number=record_number)
+    f.close()
+    return obs_data
+
+
+# assign satellite identifier
+def get_WMO_satellite_ID(filename):
```

Comment:
this is also an attribute the file:  `f.attrs['leo']`

but we should request from AER to add the WMO satellite ID it is pretty standard

Reply:
thank you for the comment. Resolved!